src/picologging/formatter.cxx

```diff
@@ -170,7 +170,7 @@ PyObject* Formatter_format(Formatter *self, PyObject *record){
             }
             PyUnicode_Append(&result, logRecord->excText);
         }
-        if (logRecord->stackInfo != Py_None && logRecord->stackInfo != Py_False){
+        if (logRecord->stackInfo != Py_None && logRecord->stackInfo != Py_False && PyUnicode_GET_LENGTH(logRecord->stackInfo) > 0) {
```

Comment:
I added this to make this condition actually match the Python condition (which is just `if record.stack_info`)